test/org/parosproxy/paros/core/scanner/VariantMultipartFormParametersUnitTest.java

```diff
@@ -0,0 +1,243 @@
+	
+	private static final String CRLF = "\r\n";
+	private static final String DEFAULT_PARAM_CONTENT = "anonymous" + CRLF + "anonymous2";
+	private static final String DEFAULT_FILE_NAME = "file.ext";
+	private static final String DEFAULT_CONTENT_TYPE = "text/plain";
+	private static final String DEFAULT_FILE_PARAM_CONTENT = "contents of the file";
+
+	@Test
+	public void shouldHaveParametersListEmptyByDefault() {
+		// Given
+		VariantMultipartFormParameters variant = new VariantMultipartFormParameters();
+		// When
+		List<NameValuePair> parameters = variant.getParamList();
+		// Then
+		assertThat(parameters, is(empty()));
+	}
+
+	@Test(expected = UnsupportedOperationException.class)
+	public void shouldNotAllowToModifyReturnedParametersList() {
+		// Given
+		VariantMultipartFormParameters variant = new VariantMultipartFormParameters();
+		// When
+		variant.getParamList().add(new NameValuePair(NameValuePair.TYPE_POST_DATA, "name", "fred", 1));
+		// Then = UnsupportedOperationException
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void shouldFailToExtractParametersFromUndefinedMessage() {
+		// Given
+		VariantMultipartFormParameters variant = new VariantMultipartFormParameters();
+		HttpMessage undefinedMessage = null;
+		// When
+		variant.setMessage(undefinedMessage);
+		// Then = IllegalArgumentException
+	}
+
+	@Test
+	public void shouldExtractParametersFromAllParts() {
+		// Given
+		VariantMultipartFormParameters variant = new VariantMultipartFormParameters();
+		// When
+		variant.setMessage(createMessage());
+		// Then
+		assertThat(variant.getParamList().size(), is(equalTo(4)));
+		assertThat(variant.getParamList().get(0).getName(), is(equalTo("person")));
+		assertThat(variant.getParamList().get(0).getValue(), is(equalTo(DEFAULT_PARAM_CONTENT)));
+		assertThat(variant.getParamList().get(1).getName(), is(equalTo("somefile")));
+		assertThat(variant.getParamList().get(1).getValue(), is(equalTo(DEFAULT_FILE_PARAM_CONTENT)));
+		assertThat(variant.getParamList().get(2).getName(), is(equalTo("somefile")));
+		assertThat(variant.getParamList().get(2).getValue(), is(equalTo(DEFAULT_FILE_NAME)));
+		assertThat(variant.getParamList().get(3).getName(), is(equalTo("somefile")));
+		assertThat(variant.getParamList().get(3).getValue(), is(equalTo(DEFAULT_CONTENT_TYPE)));
+	}
+
+	@Test
+	public void shouldInjectParamValueModificationInGeneralParam() {
+		// Given
+		VariantMultipartFormParameters variant = new VariantMultipartFormParameters();
+		HttpMessage message = createMessage();
+		String paramName = "person";
+		String newValue = "injected";
+		variant.setMessage(message);
+		// When
+		variant.setParameter(message,
+				new NameValuePair(NameValuePair.TYPE_POST_DATA, paramName, DEFAULT_PARAM_CONTENT, 1),
+				paramName, newValue);
+		StringBuilder testBodyBuilder = new StringBuilder(320);
+		testBodyBuilder.append("--------------------------d74496d66958873e").append(CRLF);
```

Comment:
How about changing `createMessage` to allow to pass the 4 fields and call that with the corresponding values? It would reduce duplication and still check the correct structure.

Reply:
Yup I like that.